src/array.cr

```diff
@@ -2029,6 +2029,31 @@ class Array(T)
     @capacity - @offset_to_buffer
   end
 
+  private def rewind
+    root_buffer.copy_from(@buffer, @size)
+    shift_buffer_by -@offset_to_buffer
+  end
+
+  # Enusures that the internal buffer has at least `capacity` elements.
+  def ensure_capacity(capacity : Int32) : self
+    rewind
+
+    if capacity >= @size
+      resize_to_capacity capacity
+    end
+    self
+  end
+
+  # Reduces the internal buffer to exactly fit the number of elements in the
+  # array, plus `extra` elements.
+  def trim_to_size(extra : Int32 = 0) : self
```

Comment:
This extra argument is pretty confusing. Trim to size 2 looks like it will trim it to size 2. I would rename extra to "plus", and make it a named argument. Then it will read like "trim to size plus 2".

Reply:
Hmm I'm not sure I agree on the word. `extra` is fully correct English, I slightly prefer that one. "with 2 extra elements".
But yes, definitely would need to remove the ability to pass this as a positional argument :+1:

Reply:
Yes, it's mainly that as a positional argument a reader might be confused. The named argument could still be extra, but then it doesn't read like an English sentence. "trim to size extra 2" vs "trim to size plus 2". Maybe there's another word or set of words we could use.

Reply:
Vote:

 ❤️   `a.trim_to_size(plus: 2)` 
 😄 `a.trim_to_size(extra: 2)` 
 🎉  `a.trim_to_size(with_extra: 2)`

I can't think of another relevant names.

Reply:
I dislike the idea of naming a parameter `with_anything`. My impression was that the "with" is implied by the opening of the list of arguments -- of course you're calling the method with something. I was going to make an argument that by that logic you could add `with_` to any parameter whatsoever, but actually I didn't find any examples, so it could just be that I'm wrong here.

Reply:
Maybe `bonus`? :)

Reply:
Or more technical `additive`.